manifests/server/foreman.pp

```diff
@@ -0,0 +1,33 @@
+# @summary Set up Foreman integration
+# @api private
+class puppet::server::foreman (
+  Boolean $katello = false,
+) {
+  if $katello {
+    include certs::puppet
+    Class['certs::puppet'] -> Class['puppetserver_foreman']
+
+    $ssl_ca = $certs::puppet::ssl_ca_cert
+    $ssl_cert = $certs::puppet::client_cert
+    $ssl_key = $keys::puppet::client_key
+  } else {
+    $ssl_ca = $puppet::server::ssl_ca_cert
+    $ssl_cert = $puppet::server::ssl_cert
+    $ssl_key = $puppet::server::ssl_cert_key
```

Comment:
I am wondering if this should now be:

```
    $ssl_ca = pick($puppet::server::foreman_ssl_ca, $puppet::server::ssl_ca_cert)
    $ssl_cert = pick($puppet::server::foreman_ssl_cert, $puppet::server::ssl_cert)
    $ssl_key = pick($puppet::server::foreman_ssl_key, $puppet::server::ssl_cert_key)
```

To prevent a user from setting `katello=true` and then overriding it by setting those parameters and ending up in a weird state given the `katello=true` is an explicit request. Then down below it would become:

```
    ssl_ca           => $ssl_ca,
    ssl_cert         => $ssl_cert,
    ssl_key          => $ssl_key,
```

Reply:
I debated this. On the other hand, we have the parameter and if a user does want to override it ... Still not 100% sure where I land on it.

Reply:
I think my proposed change still allows overriding it, users would just set `katello=false` and provide them and prevent the whole need to generate unnecessary certs and relationships.